### PR TITLE
basic parsing of bgs...

### PR DESCRIPTION
**Pull request by prsauer**
Adds `BattlegroundCombat.ts`, a new segment parser for bg data -- this is skeletally similar to CombatData.ts but there are so many caveats/edge cases for bgs combining them would be fairly messy

Adds `LogParserSpec` typing for the parser's event emitter. This should preclude the need to cast in many places.

Adds a sample bg blitz log and test. The log has been decimated 1:3 to save file size but is still quite large.

Extends `logger.ts` to help debugging more complex logs in development

### CODE REVIEW

**Comment by atinylittleshell** [comment]
hmm lock file shouldn't have conflicts here. did you do a fresh "npm install"? "npm ci" would minimize lock file changes.

File: packages/parser/src/BattlegroundData.ts [hunk 1]
```diff
@@ -0,0 +1,414 @@
...
+      getUnitReaction(event.srcUnitFlags) !== getUnitReaction(event.destUnitFlags) &&
+      getUnitType(event.srcUnitFlags) === CombatUnitType.Player &&
+      getUnitType(event.destUnitFlags) === CombatUnitType.Player;
+
+    if (isSignificantEvent) {
+      this.inferredCombatantIds.add(event.srcUnitId);
+      this.inferredCombatantIds.add(event.destUnitId);
+    }
+
+    switch (event.logLine.event) {
```
Comment (atinylittleshell) [inline packages/parser/src/BattlegroundData.ts:125]: a lot of the logic should be consistent with the other combat class. might be good to extract to a base class and reuse?

**Comment by prsauer** [comment]
> hmm lock file shouldn't have conflicts here. did you do a fresh "npm install"? "npm ci" would minimize lock file changes.

I did `npm i` but I think the changes it did are pretty reasonable. The internal package dep for our project updated and eventemitter3 got an actual regex for version instead of *

File: packages/parser/src/BattlegroundData.ts (hunk 1 above)
Comment (prsauer) [reply packages/parser/src/BattlegroundData.ts:125]: yeah I need to have a think through this. there's definitely going to be some common code but also a lot of code that *looks* common but has super subtle diffs

**Review by atinylittleshell — APPROVED** [review]
